index.js

```diff
@@ -72,22 +72,22 @@ module.exports = {
     "simple-import-sort/imports": "error",
     "simple-import-sort/exports": "error",
     "sort-destructure-keys/sort-destructure-keys": "error",
-    /**
-     * List of sonarjs rules we turned off after release of eslint-plugin-sonarjs v2
-     * Some rules are too restrictive for common use cases,
-     * and others are already covered by other plugins used here.
-     */
-    "sonarjs/different-types-comparison": "off",
```

Comment:
I figured converting to a warning will probably not be enough to enforce a different pattern in cases when we want to discourage casts, as warnings have a tendancy to survive for a pretty long time 😄 . So folks would need to explicitly disable the rule if they are certain that they need to compare objects of different types, which is more intentional.